versioned_docs/version-2020fa/lecture7.md

```diff
@@ -3,4 +3,412 @@ id: lecture7
+   parent component
+6. Connect component to the backend (more on this in Lecture 8!)
+
+## Filterable Product Table Example
+
+This section contains the code from the live demo presented during class. Watch the
+lecture video linked at the top for an explanation of the code, intended to
+teach how to think in the React development paradigm.
+
+```tsx title="App.tsx (root component)"
+import React from 'react';
+import FilterableProductTable from './FilterableProductTable';
+
+const PRODUCTS = [
+  {
+    category: 'Sporting Goods',
+    price: '$49.99',
+    stocked: true,
+    name: 'Football',
+  },
+  {
+    category: 'Sporting Goods',
+    price: '$9.99',
+    stocked: true,
+    name: 'Baseball',
+  },
+  {
+    category: 'Sporting Goods',
+    price: '$29.99',
+    stocked: false,
+    name: 'Basketball',
+  },
+  {
+    category: 'Electronics',
+    price: '$99.99',
+    stocked: true,
+    name: 'iPod Touch',
+  },
+  {
+    category: 'Electronics',
+    price: '$399.99',
+    stocked: false,
+    name: 'iPhone 5',
+  },
+  {
+    category: 'Electronics',
+    price: '$199.99',
+    stocked: true,
+    name: 'Nexus 7',
+  },
+];
+
+const App = () => (
+  <div className="App">
+    <FilterableProductTable products={PRODUCTS} />
+  </div>
+);
+
+export default App;
+```
+
+```tsx title="FilterableProductTable.tsx"
+import React, { ChangeEvent, useState } from 'react';
+import ProductTable, { Product } from './Starter';
+
+type TableProps = {
+  readonly products: Product[];
+};
+
+type SearchProps = {
+  readonly filterText: string;
+  readonly inStockOnly: boolean;
+  readonly handleFilterTextChange: (e: ChangeEvent<HTMLInputElement>) => void;
+  readonly handleCheckBoxChange: (e: ChangeEvent<HTMLInputElement>) => void;
+};
+
+const SearchBar = ({
+  filterText,
+  inStockOnly,
+  handleFilterTextChange,
+  handleCheckBoxChange,
+}: SearchProps) => (
+  <form>
+    <input
+      type="text"
+      placeholder="Search..."
+      value={filterText}
+      onChange={handleFilterTextChange}
+    />
+    <p>
+      <input
+        type="checkbox"
+        checked={inStockOnly}
+        onChange={handleCheckBoxChange}
+      />{' '}
+      Only show products in stock
+    </p>
+  </form>
+);
+
+const FilterableProductTable = ({ products }: TableProps) => {
+  const [filterText, setFilterText] = useState('');
+  const [inStockOnly, setInStockOnly] = useState(false);
+
+  const handleFilterTextChange = (e: ChangeEvent<HTMLInputElement>) =>
+    setFilterText(e.target.value);
+  const handleCheckBoxChange = (e: ChangeEvent<HTMLInputElement>) =>
+    setInStockOnly(e.target.checked);
+
+  return (
+    <div>
+      <SearchBar
+        filterText={filterText} // states passed as prop to SearchBar
+        inStockOnly={inStockOnly} // states passed as prop to SearchBar
+        handleFilterTextChange={handleFilterTextChange} // only step 5
+        handleCheckBoxChange={handleCheckBoxChange} // only step 5
```

Comment:
rest of the comments are fine but maybe remove this or change it to `pass down handler functions to update search state`.